src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientTest.cs

```diff
@@ -205,6 +205,15 @@ public async Task GetContentAsync_ErrorStatusCode_ExpectedExceptionThrown(bool w
             }
         }
 
+        [Fact]
+        [OuterLoop("Slow - Negative connection test")]
+        public async Task GetContentAsync_WhenCanNotConnect_ExceptionContainsHostInfo()
+        {
+            using var client = new HttpClient(new SocketsHttpHandler());
+            HttpRequestException ex = await Assert.ThrowsAsync<HttpRequestException>(() => client.GetStreamAsync("http://localhost:4242"));
```

Comment:
I'm not sure I'm comfortable to connect to random port and assume it is closed. Anything above 1024 should be free game AFAIK and test suite binds on random ports. Would it make sense to Bind() on anonymous port to get porte reserved and then Don't call Listen/Accept on it? I think that would make the test more predictable.